hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndex.java

```diff
@@ -154,6 +154,14 @@ public boolean requiresTagging(WriteOperationType operationType) {
   public void close() {
   }
 
+  /***
+   * Updates index metadata of the given table and instant if needed.
+   * @param table The committed table.
+   * @param hoodieInstant The instant to commit.
+   */
+  public void commitIndexMetadataIfNeeded(HoodieTable table, String hoodieInstant) {
+  }
+
```

Comment:
Do we need a new interface? Can we just make the metadata right in `updateLocation` ?

Reply:
updateLocation is pre-commit operation, that's the reason i haven't used it.I was thinking we can use commitIndexMetadta interface for post commit operation.

Reply:
let me know your thoughts on this. idea is need to create commit marker file once clustering operation gets committed.

Reply:
One thing we can do if we don't want to use CommitIndexMetadta interface is to let loadMetadta function create commit marker file , when reader ask for updated metadata, Commit marker recovery function(recommitMetadataFile) will decide if this metadata file is latest or not and create marker file if it is latest.

Only cost of this will be listing all filegroups for that partition and comparing against metadata file. this will be done only once when commit file is not present.

Reply:
So in no way, the `updateLocation` can ensure the automcity of index metadata and the instant commit, we can fix that in similar way like what we do in `loadMetadata`

Reply:
Yeah in updateLocation itself we can't ensure atomicity of clustering operation. 
In UpdateLocation we are already creating metadata through (saveMetadata), we can treat that metadata as in-flight metadata, and loadMetadata will decide if that metadata can be committed or not. I think we can use this mechanism to ensure consistency of metadata.

Reply:
Cool, then we can get rid of the in-consistency and also the method `commitIndexMetadataIfNeeded`.

Reply:
Yeah sure , let me add the patch

Reply:
done , removed the commitIndex interface. please review